blocks/i-lens/i-lens.js

```diff
@@ -0,0 +1,216 @@
+  const rowDiv = document.createElement('div');
+  rowDiv.className = 'row align-items-center';
+
+  if (dropdowns) {
+    const dropdownsDiv = document.createElement('div');
+    dropdownsDiv.className = 'dropdowns col';
+    dropdowns.forEach((dropdownValue) => {
+      const dropDownEle = createDropdown(dropdownValue);
+      dropdownsDiv.appendChild(dropDownEle);
+    });
+
+    rowDiv.appendChild(dropdownsDiv);
+    document.addEventListener('click', (event) => {
+      closeAllDropDowns(event.target);
+    });
+  }
+
+  header.appendChild(rowDiv);
+  ilensContainer.appendChild(header);
+}
+
+// eslint-disable-next-line no-unused-vars
+function addStocksData(ilensBody, dropdown = 'default') {
+  ilensBody.textContent = '';
+  fetchData(`${getHostUrl()}/scripts/mock-ilensdata.json`, async (error, ilensData = []) => {
+    const { tableData } = ilensData[dropdown].body;
+    // Sort the tableData based on the Operating Revenue Qtr (SR_Q) in descending order
+    tableData.sort((a, b) => b[6] - a[6]);
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < 5; i++) {
+      const [stockName, ...revenues] = tableData[i];
+
+      const box = document.createElement('div');
+      box.classList.add('box');
+
+      const heading = document.createElement('h3');
+      heading.textContent = stockName;
+      box.appendChild(heading);
+
+      const list = document.createElement('ul');
+      revenues.slice(5, 14).forEach((revenue, index) => {
+        const listItem = document.createElement('li');
+
+        // eslint-disable-next-line no-shadow
+        const label = document.createElement('label');
+        // eslint-disable-next-line max-len
+        label.appendChild(document.createTextNode(ilensData[dropdown].body.tableColumns[index + 6].name));
+        listItem.appendChild(label);
+
+        // eslint-disable-next-line no-shadow
+        const span = document.createElement('span');
+        span.classList.add('value');
+        if (revenue.toLocaleString().endsWith('%') && parseFloat(revenue) > 0) {
+          span.classList.add('green');
+        }
+        span.textContent = revenue.toLocaleString();
+        listItem.appendChild(span);
+
+        list.appendChild(listItem);
+      });
+      box.appendChild(list);
+
+      // Append the box to the document body or any other container
+      ilensBody.appendChild(box);
+    }
+  });
+}
+function addDiscoverLink(ilensBody, discoverLink) {
+  if (discoverLink) {
+    const div = document.createElement('div');
+    div.className = 'text-center discover-more';
+    const anchor = document.createElement('a');
+    anchor.href = discoverLink; // Set the href to your discoverLink variable
+    anchor.className = 'link-color';
+    anchor.target = '_blank'; // Ensures the link opens in a new tab
+    anchor.textContent = 'Discover More '; // Add the text content
+    const icon = document.createElement('i');
+    icon.className = 'icon-up-arrow icon ';
```

Comment:
arrow icon is missing on UI
<img width="544" alt="image" src="https://github.com/aemsites/icicidirect/assets/46780200/194bfa34-2a23-426d-be0c-18a72d37d5ae">

Reply:
@ravverma updated